fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -489,14 +492,27 @@ private void handleShowMigrations() throws AnalysisException {
     }
 
     // Show databases statement
-    private void handleShowDb() throws AnalysisException {
+    private void handleShowDb() throws AnalysisException, DdlException {
         ShowDbStmt showDbStmt = (ShowDbStmt) stmt;
         List<List<String>> rows = Lists.newArrayList();
-        List<String> dbNames = ctx.getGlobalStateMgr().getClusterDbNames(ctx.getClusterName());
+        List<String> dbNames = new ArrayList<>();
+
+        if (showDbStmt.getCatalogName() == null) {
+            dbNames = ctx.getGlobalStateMgr().getClusterDbNames(ctx.getClusterName());
+        } else {
+            String db = showDbStmt.getCatalogName();
+            MetadataMgr metadataMgr = GlobalStateMgr.getCurrentState().getMetadataMgr();
```

Comment:
todo: add metadataMgr to field in ShowExecutor

Reply:
> todo: add metadataMgr as field in ShowExecutor

done